### PR TITLE
support shared server for multi apiserver, workloadcontroller, etcd

### PR DESCRIPTION
**Pull request by sonyafenge**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:
support shared server for multi apiserver, workloadcontroller, etcd
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -43,6 +43,7 @@ NODE_LABELS="${KUBE_NODE_LABELS:-}"
 WINDOWS_NODE_LABELS="${WINDOWS_NODE_LABELS:-}"
 APISERVERS_EXTRA_NUM=${APISERVERS_EXTRA_NUM:-0}
 WORKLOADCONTROLLER_EXTRA_NUM=${WORKLOADCONTROLLER_EXTRA_NUM:-0}
+<<<<<<< ddcd1d53ee254d98641dde47c58edf78a908ceb3
```
Comment (Sindica) [inline cluster/gce/config-default.sh:46]: Resolve conflict here

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -55,6 +56,14 @@ WORKLOADCONTROLLER_MIN_CPU_ARCHITECTURE=${WORKLOADCONTROLLER_MIN_CPU_ARCHITECTUR
...
 WORKLOADCONTROLLER_DISK_SIZE=${WORKLOADCONTROLLER_DISK_SIZE:-${MASTER_DISK_SIZE}}
 WORKLOADCONTROLLER_ROOT_DISK_SIZE=${WORKLOADCONTROLLER_ROOT_DISK_SIZE:-${MASTER_ROOT_DISK_SIZE}}
+=======
+SHARE_PARTITIONSERVER=${SHARE_PARTITIONSERVER:-false}
+PARTITIONSERVER_SIZE=${PARTITIONSERVER_SIZE:-${MASTER_SIZE}}
+PARTITIONSERVER_MIN_CPU_ARCHITECTURE=${PARTITIONSERVER_MIN_CPU_ARCHITECTURE:-${MASTER_MIN_CPU_ARCHITECTURE:-}} # To allow choosing better architectures.
+PARTITIONSERVER_DISK_TYPE=${PARTITIONSERVER_DISK_TYPE:-${MASTER_DISK_TYPE}}
+PARTITIONSERVER_DISK_SIZE=${PARTITIONSERVER_DISK_SIZE:-${MASTER_DISK_SIZE}}
+PARTITIONSERVER_ROOT_DISK_SIZE=${PARTITIONSERVER_ROOT_DISK_SIZE:-${MASTER_ROOT_DISK_SIZE}}
+>>>>>>> support shared server for apiserver, workloadcontroller, etcd
```
Comment (Sindica) [inline cluster/gce/config-default.sh:66]: And here

File: cluster/gce/config-test.sh [hunk 3]
```diff
@@ -43,6 +43,7 @@ NODE_LABELS="${KUBE_NODE_LABELS:-}"
 WINDOWS_NODE_LABELS="${WINDOWS_NODE_LABELS:-}"
 APISERVERS_EXTRA_NUM=${APISERVERS_EXTRA_NUM:-0}
 WORKLOADCONTROLLER_EXTRA_NUM=${WORKLOADCONTROLLER_EXTRA_NUM:-0}
+<<<<<<< ddcd1d53ee254d98641dde47c58edf78a908ceb3
```
Comment (Sindica) [inline cluster/gce/config-test.sh:46]: Resolve conflict

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (sonyafenge) [reply cluster/gce/config-default.sh]: just updated and resolved the conflict.

File: cluster/gce/config-default.sh (hunk 2 above)
Comment (sonyafenge) [reply cluster/gce/config-default.sh]: just updated and resolved the conflict.

File: cluster/gce/config-test.sh (hunk 3 above)
Comment (sonyafenge) [reply cluster/gce/config-test.sh]: just updated and resolved the conflict.

**Comment by sonyafenge** [comment]
Add arg for kube-up.sh and start-kubemark.sh:  SHARE_PARTITIONSERVER, if you want to share all extra partitionservers, then set SHARE_PARTITIONSERVER=true; if you want to create dedicated vms for all your extra partitionserver, then set SHARE_PARTITIONSERVER=false

**Pull request merged**